### PR TITLE
fix: metadata padding

### LINKED ISSUE #35: Padding is missing when adding metadata
When adding metadata the Bee implementation it also adds padding so that the number of bytes for the metadata is rounded up to the nearest integer multiplier of the obfuscation key size.

Currently this padding is missing from mantaray-js, therefore the created manifests does not result in the same content hash, although the stored information is the same.

### PR DESCRIPTION
**Pull request by agazso**
Fixes #35. Adds a test to verify that the content hash is exactly the same as what Bee generates.

### CODE REVIEW

File: src/node.ts [hunk 1]
```diff
@@ -99,10 +100,25 @@ export class MantarayFork {
       const metadataBytes = new TextEncoder().encode(jsonString)
 
       // pad JSON bytes if necessary -> the encryptDecrypt handles if the data has no key length
+      const metadataSizeWithSize = metadataBytes.length + nodeForkSizes.metadata
+      let padding = new Uint8Array(0)
+      if (metadataSizeWithSize < nodeHeaderSizes.obfuscationKey) {
```
Comment (nugaon) [inline src/node.ts:105]: `Expected blank line before this statement`

below the same problem

File: src/node.ts [hunk 2]
```diff
@@ -99,10 +100,25 @@ export class MantarayFork {
       const metadataBytes = new TextEncoder().encode(jsonString)
 
       // pad JSON bytes if necessary -> the encryptDecrypt handles if the data has no key length
+      const metadataSizeWithSize = metadataBytes.length + nodeForkSizes.metadata
```
Comment (nugaon) [inline src/node.ts:103]: remove the comment above

**Review by nugaon — APPROVED** [review]
I would extract this logic into a separate `createMetadataPadding` function, but anyway after correcting the comments below, LGTM

**Comment by agazso** [comment]
Please review it again. I made the changes you asked for, also I updated to use Bee 1.7.0 with the new bee-factory and bee-js 5.0.0.

**Comment by agazso** [comment]
Also fixes #38

**Review by nugaon — APPROVED** [review]
as the test shows, it is an improvement in order to be compatible with the Bee client, but I commented below, I don't think this is the proper way to pad metadata. It adds new line characters in order to not break the stringified JSON structure on load, but it does not carry any information. The deserializer should handle this in an efficient way as this library did so far. I would create a ticket about it in Bee but to be honest I would prefer more to upgrade to a newer version of mantaray already.

File: src/node.ts [hunk 3]
```diff
@@ -98,11 +119,12 @@ export class MantarayFork {
       const jsonString = JSON.stringify(this.node.getMetadata)
       const metadataBytes = new TextEncoder().encode(jsonString)
 
-      // pad JSON bytes if necessary -> the encryptDecrypt handles if the data has no key length
+      const metadataSizeWithSize = metadataBytes.length + nodeForkSizes.metadata
+      const padding = this.createMetadataPadding(metadataSizeWithSize)
 
-      const metadataBytesSize = toBigEndianFromUint16(metadataBytes.length)
+      const metadataBytesSize = toBigEndianFromUint16(metadataBytes.length + padding.length)
```
[the remark this answers is not in the document]
Comment (agazso) [reply src/node.ts:125]: Here the goal was to replicate the implementation in Bee and this is the exact same thing.

**Pull request merged**